tools/java/BUILD

```diff
@@ -19,8 +19,21 @@ genrule(
     visibility = ["//tools:__pkg__"],
 )
 
+# Generate a BUILD.bazel file to put in the repo, as this one references targets which don't exist in a Bazel distribution.
+# Note: We need to call it BUILD.bazel not BUILD because otherwise it conflicts with this file.
+genrule(
+    name = "generated_BUILD",
+    srcs = ["BUILD.for_buildtools"],
+    outs = ["BUILD.bazel"],
+    cmd = "cp $< $@",
+)
```

Comment:
I'd prefer if BUILD.tools file is added to the repo and used in the filegroup, like it's done in other directories (tools/jdk, tools/java/runfiles).

java_stub_template.txt needed genrule with cp, but in general I'd prefer it's avoided.

Reply:
Done!

Reply:
You also need to add BUILD.tools to srcs filegroup in tools/java/BUILD